pipeline/src/pipeline.rs

```diff
@@ -681,10 +682,9 @@ impl<T: FieldElement> Pipeline<T> {
 
     fn maybe_write_constants(&self, constants: &[(String, Vec<T>)]) -> Result<(), Vec<String>> {
         if let Some(path) = self.path_if_should_write(|name| format!("{name}_constants.bin"))? {
-            write_polys_file(
-                &mut BufWriter::new(&mut fs::File::create(path).unwrap()),
-                constants,
-            );
+            let mut file = BufWriter::new(fs::File::create(path).unwrap());
+            write_polys_file(&mut file, constants);
+            file.flush().unwrap();
```

Comment:
Why not keep it as a separate function? These 3 lines are duplicated in other places

Reply:
Placed it in a different function.